core/domain/fs_domain.py

```diff
@@ -325,7 +325,8 @@ def commit(self, filepath, raw_bytes, mimetype=None):
             raw_bytes: str. The content to be stored in the file.
             mimetype: str. The content-type of the file.
         """
-        raw_bytes = python_utils.convert_to_bytes(raw_bytes)
+        if not mimetype or mimetype != 'application/octet-stream':
```

Comment:
Let's not use the octet-stream mimetype -- see above.

Reply:
I don't think we can store binary files in any other way than this. @BenHenning is it possible to avoid octet-stram to store protobuf encoded data?

Reply:
Oh sorry, missed that this was for GCS. I'm not 100% certain, but I think it's probably ok in this case.

Reply:
Keeping this as it is.

Reply:
Actually, reading this more carefully, I'm a bit confused. Isn't the if condition always true?

Also why do we need the if condition anyway?

Reply:
Not neccessarily, filetype of [images](https://github.com/oppia/oppia/blob/b02429ffaca99c470e6a329185ea43515eb3c0b7/core/domain/fs_services.py#L69) is different.

Reply:
Actually, my bad, I meant that the first condition is a subset of the second. Couldn't all this be replaced with "if mimetype != 'application/octet-stream'"?

Reply:
Ah, yes. Python doesn't raise exception on comparing None. My bad.